components/tidb_query_vec_expr/src/impl_time.rs

```diff
@@ -719,6 +725,40 @@ mod tests {
         test_null_case(ScalarFuncSig::MicroSecond);
     }
 
+    #[test]
+    fn test_time_to_sec() {
+        let cases: Vec<(&str, i8, i64)> = vec![
+            ("31 11:30:45", 0, 2719845),
+            ("11:30:45.123345", 3, 41445),
+            ("-11:30:45.1233456", 0, -41445),
+            ("272:59:59.14", 0, 982799),
+        ];
+
+        for (arg, fsp, s) in cases {
+            let duration =
+                Some(Duration::parse(&mut EvalContext::default(), arg.as_bytes(), fsp).unwrap());
+            let test_case_func = |sig, res| {
+                let output = RpnFnScalarEvaluator::new()
+                    .push_param(duration)
+                    .evaluate::<Int>(sig)
+                    .unwrap();
+                assert_eq!(output, Some(res));
+            };
+            test_case_func(ScalarFuncSig::TimeToSec, s);
+        }
+
+        // test NULL case
+        let test_null_case = |sig| {
+            let output = RpnFnScalarEvaluator::new()
+                .push_param(None::<Duration>)
+                .evaluate::<Int>(sig)
+                .unwrap();
+            assert_eq!(output, None);
+        };
+
+        test_null_case(ScalarFuncSig::Hour);
```

Comment:
Why are we testing with `ScalarFuncSig::Hour`? And also `time_to_sec` needs a parameter.

Reply:
This line is redundant. I deleted it.

Reply:
It was mistake. I rewrited test.